File: iosApp/iosApp/Screens/AccessibilitySettingsScreen.swift
import SwiftUI

/**
 * Accessibility settings screen for iOS
 */
struct AccessibilitySettingsScreen: View {
    @Environment(\.presentationMode) var presentationMode
    @StateObject private var accessibilityHelper = AccessibilityHelper()
    
    // Local state for settings (in a real app, this would be connected to the Kotlin shared module)
    @State private var textScale: Double = 1.0
    @State private var isHighContrastEnabled = false
    @State private var isLargeTouchTargetsEnabled = true
    @State private var isReducedMotionEnabled = false
    
    var body: some View {
        NavigationView {
            ScrollView {
                VStack(alignment: .leading, spacing: 24) {
                    
                    // Text Size Section
                    VStack(alignment: .leading, spacing: 16) {
                        Text("Text Size")
                            .font(.headline)
                            .accessibilityAddTraits(.isHeader)
                        
                        Text("Adjust text size for better readability")
                            .font(.caption)
                            .foregroundColor(.secondary)
                        
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Current size: \(Int(textScale * 100))%")
                                .font(.caption)
                            
                            Slider(
                                value: $textScale,
                                in: AccessibilityConstants.minTextScale...AccessibilityConstants.maxTextScale,
                                step: 0.1
                            ) {
                                Text("Text Size")
                            } minimumValueLabel: {
                                Text("Small")
                                    .font(.caption2)
                            } maximumValueLabel: {
                                Text("Large")
                                    .font(.caption2)
                            }
                            .accessibilityLabel("Text size slider")
                            .accessibilityValue("\(Int(textScale * 100)) percent")
                        }
                    }
                    .padding()
                    .background(Color(.systemBackground))
                    .cornerRadius(12)
                    .shadow(radius: 2)
                    
                    // High Contrast Section
                    VStack(alignment: .leading, spacing: 16) {
                        HStack {
                            VStack(alignment: .leading, spacing: 4) {
                                Text("High Contrast")
                                    .font(.headline)
                                
                                Text("Increase contrast for better visibility")
                                    .font(.caption)
                                    .foregroundColor(.secondary)
                            }
                            
                            Spacer()
                            
                            Toggle("", isOn: $isHighContrastEnabled)
                                .accessibilityLabel("High contrast")
                                .accessibilityValue(isHighContrastEnabled ? "enabled" : "disabled")
                        }
                    }
                    .padding()
                    .background(Color(.systemBackground))
                    .cornerRadius(12)
                    .shadow(radius: 2)
                    
                    // Large Touch Targets Section
                    VStack(alignment: .leading, spacing: 16) {
                        HStack {
                            VStack(alignment: .leading, spacing: 4) {
                                Text("Large Touch Targets")
                                    .font(.headline)
                                
                                Text("Make buttons and interactive elements larger")
                                    .font(.caption)
                                    .foregroundColor(.secondary)
                            }
                            
                            Spacer()
                            
                            Toggle("", isOn: $isLargeTouchTargetsEnabled)
                                .accessibilityLabel("Large touch targets")
                                .accessibilityValue(isLargeTouchTargetsEnabled ? "enabled" : "disabled")
                        }
                    }
                    .padding()
                    .background(Color(.systemBackground))
                    .cornerRadius(12)
                    .shadow(radius: 2)
                    
                    // Reduced Motion Section
                    VStack(alignment: .leading, spacing: 16) {
                        HStack {
                            VStack(alignment: .leading, spacing: 4) {
                                Text("Reduce Motion")
                                    .font(.headline)
                                
                                Text("Minimize animations and transitions")
                                    .font(.caption)
                                    .foregroundColor(.secondary)
                            }
                            
                            Spacer()
                            
                            Toggle("", isOn: $isReducedMotionEnabled)
                                .accessibilityLabel("Reduced motion")
                                .accessibilityValue(isReducedMotionEnabled ? "enabled" : "disabled")
                        }
                    }
                    .padding()
                    .background(Color(.systemBackground))
                    .cornerRadius(12)
                    .shadow(radius: 2)
                    
                    // System Accessibility Status
                    VStack(alignment: .leading, spacing: 16) {
                        Text("System Accessibility Status")
                            .font(.headline)
                            .accessibilityAddTraits(.isHeader)
                        
                        VStack(alignment: .leading, spacing: 8) {
                            AccessibilityStatusRow(
                                title: "VoiceOver",
                                isEnabled: accessibilityHelper.isVoiceOverRunning
                            )
                            
                            AccessibilityStatusRow(
                                title: "Switch Control",
                                isEnabled: accessibilityHelper.isSwitchControlRunning
                            )
                            
                            AccessibilityStatusRow(
                                title: "Reduce Motion",
                                isEnabled: accessibilityHelper.isReduceMotionEnabled
                            )
                            
                            AccessibilityStatusRow(
                                title: "Reduce Transparency",
                                isEnabled: accessibilityHelper.isReduceTransparencyEnabled
                            )
                            
                            AccessibilityStatusRow(
                                title: "Darker System Colors",
                                isEnabled: accessibilityHelper.isDarkerSystemColorsEnabled
                            )
                        }
                    }
                    .padding()
                    .background(Color(.systemBackground))
                    .cornerRadius(12)
                    .shadow(radius: 2)
                    
                    // Reset Button
                    AccessibleButton(
                        title: "Reset to Defaults",
                        action: resetToDefaults,
                        contentDescription: "Reset all accessibility settings to default values",
                        style: .secondary
                    )
                    .padding(.top, 16)
                }
                .padding()
            }
            .navigationTitle("Accessibility Settings")
            .navigationBarTitleDisplayMode(.large)
            .navigationBarBackButtonHidden(true)
            .toolbar {
                ToolbarItem(placement: .navigationBarLeading) {
                    AccessibleIconButton(
                        systemName: "arrow.left",
                        action: {
                            presentationMode.wrappedValue.dismiss()
                        },
                        contentDescription: AccessibilityConstants.ContentDescriptions.backButton
                    )
                }
            }
        }
        .dynamicTypeSupport(textScale: CGFloat(textScale))
        .highContrastSupport()
        .reducedMotionSupport()
    }
    
    private func resetToDefaults() {
        textScale = AccessibilityConstants.defaultTextScale
        isHighContrastEnabled = false
        isLargeTouchTargetsEnabled = true
        isReducedMotionEnabled = false
    }
}

/**
 * Row component for displaying accessibility status
 */
struct AccessibilityStatusRow: View {
    let title: String
    let isEnabled: Bool
    
    var body: some View {
        HStack {
            Text(title)
                .font(.body)
            
            Spacer()
            
            Image(systemName: isEnabled ? "checkmark.circle.fill" : "circle")
                .foregroundColor(isEnabled ? .green : .gray)
                .accessibilityLabel(isEnabled ? "Enabled" : "Disabled")
        }
        .accessibilityElement(children: .combine)
        .accessibilityLabel("\(title), \(isEnabled ? "enabled" : "disabled")")
    }
}

// MARK: - Preview

struct AccessibilitySettingsScreen_Previews: PreviewProvider {
    static var previews: some View {
        AccessibilitySettingsScreen()
            .previewLayout(.sizeThatFits)
    }
}

File: androidApp/src/androidMain/kotlin/com/carecomms/android/ui/screens/AccessibilitySettingsScreen.kt
package com.carecomms.android.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.carecomms.accessibility.AccessibilityConstants
import com.carecomms.accessibility.AndroidAccessibilityPreferences
import com.carecomms.android.ui.components.AccessibleButton
import com.carecomms.android.ui.components.AccessibleIconButton
import kotlinx.coroutines.launch

/**
 * Accessibility settings screen for configuring user preferences
 */
@Composable
fun AccessibilitySettingsScreen(
    onNavigateBack: () -> Unit
) {
    val context = LocalContext.current
    val accessibilityPreferences = remember { AndroidAccessibilityPreferences(context) }
    val accessibilitySettings by accessibilityPreferences.observeAccessibilitySettings().collectAsState()
    val scope = rememberCoroutineScope()
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
    ) {
        // Top app bar
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 24.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            AccessibleIconButton(
                icon = Icons.Default.ArrowBack,
                contentDescription = AccessibilityConstants.ContentDescriptions.BACK_BUTTON,
                onClick = onNavigateBack
            )
            Spacer(modifier = Modifier.width(16.dp))
            Text(
                text = "Accessibility Settings",
                style = MaterialTheme.typography.h5,
                modifier = Modifier.semantics {
                    contentDescription = "Accessibility Settings Screen"
                }
            )
        }
        
        // Text Scale Setting
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 8.dp),
            elevation = 4.dp
        ) {
            Column(
                modifier = Modifier.padding(16.dp)
            ) {
                Text(
                    text = "Text Size",
                    style = MaterialTheme.typography.h6,
                    modifier = Modifier.padding(bottom = 8.dp)
                )
                Text(
                    text = "Adjust text size for better readability",
                    style = MaterialTheme.typography.body2,
                    color = MaterialTheme.colors.onSurface.copy(alpha = 0.7f),
                    modifier = Modifier.padding(bottom = 16.dp)
                )
                
                Text(
                    text = "Current size: ${(accessibilitySettings.textScale * 100).toInt()}%",
                    style = MaterialTheme.typography.body2,
                    modifier = Modifier.padding(bottom = 8.dp)
                )
                
                Slider(
                    value = accessibilitySettings.textScale,
                    onValueChange = { scale ->
                        scope.launch {
                            accessibilityPreferences.updateTextScale(scale)
                        }
                    },
                    valueRange = AccessibilityConstants.MIN_TEXT_SCALE..AccessibilityConstants.MAX_TEXT_SCALE,
                    steps = 10,
                    modifier = Modifier
                        .fillMaxWidth()
                        .semantics {
                            contentDescription = "Text size slider, current value ${(accessibilitySettings.textScale * 100).toInt()} percent"
                        }
                )
                
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text("Small", style = MaterialTheme.typography.caption)
                    Text("Large", style = MaterialTheme.typography.caption)
                }
            }
        }
        
        // High Contrast Setting
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 8.dp),
            elevation = 4.dp
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = "High Contrast",
                        style = MaterialTheme.typography.h6
                    )
                    Text(
                        text = "Increase contrast for better visibility",
                        style = MaterialTheme.typography.body2,
                        color = MaterialTheme.colors.onSurface.copy(alpha = 0.7f)
                    )
                }
                Switch(
                    checked = accessibilitySettings.isHighContrastEnabled,
                    onCheckedChange = { enabled ->
                        scope.launch {
                            accessibilityPreferences.updateHighContrast(enabled)
                        }
                    },
                    modifier = Modifier.semantics {
                        contentDescription = if (accessibilitySettings.isHighContrastEnabled) {
                            "High contrast enabled, tap to disable"
                        } else {
                            "High contrast disabled, tap to enable"
                        }
                    }
                )
            }
        }
        
        // Large Touch Targets Setting
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 8.dp),
            elevation = 4.dp
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = "Large Touch Targets",
                        style = MaterialTheme.typography.h6
                    )
                    Text(
                        text = "Make buttons and interactive elements larger",
                        style = MaterialTheme.typography.body2,
                        color = MaterialTheme.colors.onSurface.copy(alpha = 0.7f)
                    )
                }
                Switch(
                    checked = accessibilitySettings.isLargeTouchTargetsEnabled,
                    onCheckedChange = { enabled ->
                        scope.launch {
                            accessibilityPreferences.updateLargeTouchTargets(enabled)
                        }
                    },
                    modifier = Modifier.semantics {
                        contentDescription = if (accessibilitySettings.isLargeTouchTargetsEnabled) {
                            "Large touch targets enabled, tap to disable"
                        } else {
                            "Large touch targets disabled, tap to enable"
                        }
                    }
                )
            }
        }
        
        // Reduced Motion Setting
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 8.dp),
            elevation = 4.dp
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = "Reduce Motion",
                        style = MaterialTheme.typography.h6
                    )
                    Text(
                        text = "Minimize animations and transitions",
                        style = MaterialTheme.typography.body2,
                        color = MaterialTheme.colors.onSurface.copy(alpha = 0.7f)
                    )
                }
                Switch(
                    checked = accessibilitySettings.isReducedMotionEnabled,
                    onCheckedChange = { enabled ->
                        scope.launch {
                            accessibilityPreferences.updateReducedMotion(enabled)
                        }
                    },
                    modifier = Modifier.semantics {
                        contentDescription = if (accessibilitySettings.isReducedMotionEnabled) {
                            "Reduced motion enabled, tap to disable"
                        } else {
                            "Reduced motion disabled, tap to enable"
                        }
                    }
                )
            }
        }
        
        Spacer(modifier = Modifier.height(24.dp))
        
        // Reset to defaults button
        AccessibleButton(
            text = "Reset to Defaults",
            onClick = {
                scope.launch {
                    accessibilityPreferences.updateTextScale(AccessibilityConstants.DEFAULT_TEXT_SCALE)
                    accessibilityPreferences.updateHighContrast(false)
                    accessibilityPreferences.updateLargeTouchTargets(true)
                    accessibilityPreferences.updateReducedMotion(false)
                }
            },
            contentDescription = "Reset all accessibility settings to default values",
            modifier = Modifier.fillMaxWidth()
        )
    }
}